docs/src/main/asciidoc/credentials-provider.adoc

```diff
@@ -83,6 +83,40 @@ An alternative is to define both username and password in Vault and drop the `qu
 property from configuration. All consuming extensions do support the ability to fetch both the username
 and password from the provider, or just the password.
 
+== Time Limited Credentials
+
+A Credentials Provider may provide time limited credentials. For instance, the `vault` extension. When using
+time limited credentials, it is important to understand that consuming extensions will not have their
+credentials refreshed automatically by the Credentials Provider. Each extension must be configured to recycle its
+connections before the credentials expire.
+
+=== Datasources
+
+Datastore connections are typically pooled. When using a time limited credentials provider, the pool must be
+configured to recycle connections before each connection's credentials expire. Both JDBC and Reactive datasources
+have a `max-lifetime` configuration property that can be used to achieve this.
+
+.JDBC Datasource
+[source, properties]
+----
+quarkus.datasource.jdbc.max-lifetime=60m
+----
+
+.Reactive Datasource
+[source, properties]
+----
+quarkus.datasource.reactive.max-lifetime=60m
+----
+
+NOTE: It is the developer's responsibility to ensure that the configuration of the datasource's `max-lifetime`
+property is less than the credentials expiration time.
+
+=== RabbitMQ
+
+When using the `smallrye-reactive-messaging-rabbitmq` extension there is no configuration needed. The
```

Comment:
Out of curiosity, what's different with the RabbitMQ extension? If the CredentialsProvider has the `"expires-at` property, couldn't we use the value to configure the pool correctly?

Reply:
RabbitMQ connections supports this automatically, it has built support for dynamic credentials.

Vert.x connection pooling does not support this. I made the PR to add `max-lifetime` to the Vert.x pool and in its current state I believe it would be hard to support this type of feature. Currently the `SqlConnectionPool` doesn't allow configuration of per-connection properties. If we gain the ability to set properties (specifically something like `max lifetime`) per connection this could be implemented